tests/unit/onyxTest.js

```diff
@@ -234,13 +234,31 @@ describe('Onyx', () => {
             },
         });
 
-        return Onyx.set(ONYX_KEYS.TEST_KEY, {test1: 'test1', test2: 'test2'})
+        return Onyx.set(ONYX_KEYS.TEST_KEY, {
+            test1: {
+                test2: 'test2',
+                test3: 'test3',
+            },
+        })
             .then(() => {
-                expect(testKeyValue).toEqual({test1: 'test1', test2: 'test2'});
+                expect(testKeyValue).toEqual({
+                    test1: {
+                        test2: 'test2',
+                        test3: 'test3',
+                    },
+                });
+                return Onyx.merge(ONYX_KEYS.TEST_KEY, {
+                    test1: {
+                        test2: null,
```

Comment:
@marcaaron similar question about `undefined` values as [here](https://github.com/Expensify/react-native-onyx/pull/339#pullrequestreview-1639274665):

Should we also ignore nested `undefined` values and not overwrite the current value or should the current value be overwritten?

Reply:
We have an existing test for OnyxCache, which expects `utils.fastMerge` to do nothing, when a nested `undefined` is passed: https://github.com/Expensify/react-native-onyx/blob/eeebcb34b6a4c7b58b53e3e5ee52b84757560e48/tests/unit/onyxCacheTest.js#L312-L322

Reply:
cc @marcaaron

Reply:
Yes, I think we would just filter them out? Undefined is the _kind of_ the same as an object with no property (maybe it's arguable these are not the same or there should be some other behavior). 

WRT how Onyx should work - I think these are equivalent and both would leave the source object unchanged

```js
merge({a: '1', b: '2'}, {});
merge({a: '1', b: '2'}, {b: undefined});
```

It can work however we want - this is mainly just my opinion because I'd _prefer_ to see consistency and intentionality applied when deleting values or objects fields. If you pass `undefined` somewhere (intentionally or unintentionally) - I'm not really sure what you're trying to achieve. If you pass `null` I can see it right away. 🤷‍♂️